packages/eslint-plugin/tests/rules/no-unnecessary-type-constraint.test.ts

```diff
@@ -0,0 +1,234 @@
+import rule from '../../src/rules/no-unnecessary-type-constraint';
+import { RuleTester, getFixturesRootDir, noFormat } from '../RuleTester';
+
+const rootDir = getFixturesRootDir();
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 2018,
+    tsconfigRootDir: rootDir,
+    project: './tsconfig.json',
+  },
```

Comment:
don't need to test with type info any more

```suggestion
import { RuleTester, noFormat } from '../RuleTester';

const ruleTester = new RuleTester({
  parserOptions: {
    ecmaVersion: 2018,
  },
```